pyrokinetics/gk_code/GKInputGENE.py

```diff
@@ -78,7 +78,7 @@ def write(self, filename: PathLike, float_format: str = ""):
         super().write(filename, float_format=float_format)
 
     def is_nonlinear(self) -> bool:
-        return bool(self.data.get("nonlinear", 0))
+        return bool(self.data["general"].get("nonlinear", 0))
```

Comment:
```suggestion
        return self.data["general"].get("nonlinear", False)
```

Reply:
Should this been a boolean or a float?

Reply:
Oh, I see you've changed the other bit to be like this -- does GENE set nonlinear to 1/0 rather than T/F?

Reply:
GENE sets it to a boolean so what you have suggested is right

Reply:
It's supposed to return a bool. I only have one example GENE input file and it has `nonlinear = F`, but I forgot 1/0 are valid inputs for bools in Fortran. So I would have:

```suggestion
        return bool(self.data["general"].get("nonlinear", False))
```

Reply:
Oh god I keep changing it. This seems best